include/albatross/src/core/dataset.hpp

```diff
@@ -56,6 +56,28 @@ template <typename FeatureType> struct RegressionDataset {
   group_by(GrouperFunc grouper) const;
 };
 
+template <typename FeatureType>
+RegressionDataset<FeatureType>
+deduplicate(const RegressionDataset<FeatureType> &dataset) {
+  auto is_unique = [&](std::size_t index) -> bool {
```

Comment:
Slightly confusing function name for "appears_later".

Reply:
I see, yeah I'll change that.